### PR TITLE
MPP-3838: restore safer CSP

### PR DESCRIPTION
**Pull request by groovecoder**
Use a new EagerNonceCSPMiddleware to add nonce to the CSP and update the React app to include it in dynamic scripts.

This PR fixes MPP-3838.

How to test:
1. Open [the Relay GA4 Realtime view](https://analytics.google.com/analytics/web/#/p314403930/realtime/overview?params=_u..comparisons%3D%5B%7B%22name%22:%22Hostname%20exactly%20matches%20dev.fxprivaterelay.nonprod.cloudops.mozgcp.net%22,%22isEnabled%22:true,%22filters%22:%5B%7B%22fieldName%22:%22hostname%22,%22expressionList%22:%5B%22dev.fxprivaterelay.nonprod.cloudops.mozgcp.net%22%5D,%22isCaseSensitive%22:true%7D%5D%7D%5D%26_u..nav%3Dmaui&collectionId=user)
2. Check out this branch locally
3. In `frontend/`, `npm run watch` to build the new front-end files
4. `python manage.py runserver`
5. Visit http://127.0.0.1:8000/ and perform some page views & events
   * [ ] Check that your new events show up in the Relay GA4 Realtime view

- [x] ~l10n changes have been submitted to the l10n repository, if any.~
- [x] ~I've added a unit test to test for potential regressions of this bug.~
- [x] ~I've added or updated relevant docs in the docs/ directory.~
- [x] ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by jwhitlock — APPROVED** [review]
Thanks, looks good

File: privaterelay/middleware.py [hunk 1]
```diff
@@ -8,11 +10,42 @@
 from django.shortcuts import redirect
 
 import markus
+from csp.middleware import CSPMiddleware
 from whitenoise.middleware import WhiteNoiseMiddleware
 
 metrics = markus.get_metrics("fx-private-relay")
 
 
+CSP_NONCE_COOKIE_URLS = ["/", "/premium", "/faq", "/accounts/profile/"]
```
Comment (jwhitlock) [inline privaterelay/middleware.py:19]: Add "/phone/", and add the trailing slash for `/premium/` and `/faq/`

**Review by jwhitlock — CHANGES_REQUESTED** [review]
The tests run fine on `/accounts/profile/`, but have errors on `/faq/`.

A warning "@next/third-parties: GA dataLayer dataLayer does not exist" is raised at `GoogleAnalyticsWorkaround.tsx:103:12`. This is then followed by an error
```
Content-Security-Policy: The page’s settings blocked an inline script (script-src-elem) from being executed because it violates the following directive: “script-src 'self' https://www.google-analytics.com/ https://*.googletagmanager.com https://js.stripe.com/”
```

It doesn't happen when the trailing slash is in the path patterns.

**Comment by groovecoder** [comment]
Updated to include all url paths returned by:
```
python manage.py collectstatic
find staticfiles -type f -name 'index.html'
```
(except the `mock` urls)

File: privaterelay/middleware.py [hunk 2]
```diff
@@ -8,11 +10,61 @@
...
+    "/accounts/settings",
+    "/accounts/profile",
+    "/accounts/account_inactive",
+    "/vpn-relay-welcome",
+    "/phone/waitlist",
+    "/phone",
+    "/404",
+    "/tracker-report",
+    "/premium/waitlist",
+    "/premium",
```
Comment (jwhitlock) [inline privaterelay/middleware.py:37]: The trailing slash is needed: 
```suggestion
    "/contains-tracker-warning/",
    "/flags/",
    "/faq/",
    "/vpn-relay/waitlist/",
    "/accounts/settings/",
    "/accounts/profile/",
    "/accounts/account_inactive/",
    "/vpn-relay-welcome/",
    "/phone/waitlist/",
    "/phone/",
    "/404/",
    "/tracker-report/",
    "/premium/waitlist/",
    "/premium/",
```

**Review by jwhitlock — APPROVED** [review]
Thanks @groovecoder!

File: privaterelay/middleware.py (hunk 1 above)
Comment (Sashane969) [reply privaterelay/middleware.py]: 7f0fb8b3280f86fc5b9e94acbf77792839ca04d6

**Pull request merged**